File: Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace InteriorDesign.Entities
{
    public class Product
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [MaxLength(100)]
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        [Column(TypeName = "decimal(18, 2)")]
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
    }

}

File: ClientApp/src/models/product.ts
export class ProductModel {
    id: number;
    name: string;
    description: string;
    price: number;
    quantity: number;
    isDeleted: boolean;
    deletedAt: Date | null;

    constructor() {
        this.id = 0;
        this.name = '';
        this.description = '';
        this.price = 0;
        this.quantity = 0;
        this.isDeleted = false;
        this.deletedAt = null;
    }
}
